src/libraries/KeyLib.sol

```diff
@@ -55,17 +59,21 @@ library KeyLib {
     /// @notice Verifies a signature from `key` over a `_hash`
     function verify(Key memory key, bytes32 _hash, bytes memory signature) internal view returns (bool isValid) {
         if (key.keyType == KeyType.Secp256k1) {
-            isValid = ECDSA.recover(_hash, signature) == abi.decode(key.publicKey, (address));
+            isValid = ECDSA.tryRecover(_hash, signature) == abi.decode(key.publicKey, (address));
```

Comment:
using solady's `tryRecover` to not revert, but rather return address(0) if the signature is malformed. I prefer this over OZ's implementation which returns a bunch of other error data that we don't use

Reply:
Are you saying you need it to not revert?